name-badges/makebadges.py

```diff
@@ -0,0 +1,35 @@
+#!/bin/pyhton3
+
+import sys
+import re
+import lxml.etree as ET
+import subprocess
+#from cairosvg import svg2png
+
+namesfile = sys.argv[1]
+
+names=open(namesfile).readlines()
+
+def altersvg(file,name,role):
+    svg = open(file, mode="rb").read()
+    root = ET.fromstring(svg)
+    print(root)
+    root.findall(".//text[@inkscape:label='Name']",root.nsmap)[0][0].text=name
+    root.findall(".//text[@inkscape:label='Role']",root.nsmap)[0][0].text=role
+    return(ET.tostring(root, encoding='UTF-8'))
+
+
+
+subprocess.call(["mkdir", "tmp"])
+subprocess.call(["mkdir", "out"])
+for name in names:
+    parts = name.strip().split(":")
+    print(parts)
+    customsvg = altersvg("badge.svg",parts[0],parts[1])
+    #svg2png(bytestring=customsvg,write_t0='tmp.png')
+
+    name = parts[0]
+    writesvg = open("tmp/"+name+".svg","wb")
```

Comment:
We should use a context manager here

Reply:
and that is?

Reply:
I disagree, as I don't see it providing much benefit here. 

The main benefit of a context manager is that it would automatically clean up the file descriptor when done with it or should there be an error. But in this case, with a single write to the file as well as the fact should it error out the entire script would exit and I think python would do the GC properly... I don't see what it adds in this case

Reply:
The main benefit would be clarity of intent, though it also means that if this code were used as the basis for another project (say it were extracted to a library) then the author wouldn't need to tidy it up first.

Reply:
Considering it is literally 3 lines of code, open, write and close... the intent should be pretty obvious. Even if it were extracted to the library, unless a significant amount of changes were required which results in more lines of code for within the context manager, I still wouldn't see the point in moving over to a context manager. Considering the data to be written is generated beforehand I don't forsee the complexity of the code to increase.

One thing that could be a viable alternative is to use some form of templating engine such as Jinja2 as observing the code, the only modification made to the SVG file is setting the name and role within known areas of the file,

Reply:
It's pretty much standard to use a context manager for file operations.

Reply:
> One thing that could be a viable alternative is to use some form of templating engine such as Jinja2 as observing the code, the only modification made to the SVG file is setting the name and role within known areas of the file,

:+1:  to `jinja2`

Reply:
`jinja2` would work, but is OOS for the discussion on this line. This is about file handling, rather than how it's rendered. As @trickeydan says, it's _basically standard_ (and highly recommended) to use context managers.